### ISSUE TITLE
Support different machine types for scheduler and workers on GCP

### ISSUE THREAD

**Issue by jrbourbeau** [dask/dask-cloudprovider#390]
In https://dask.discourse.group/t/different-machine-types-for-scheduler-workers/1300 someone was asking for the ability to specify different machine types for their scheduler and workers on GCP. IIUC this isn't current implemented for `GCPCluster` but is for other managers (e.g. `EC2Cluster`). This seems like a reasonable feature request, so I wanted to log it here so we don't loose it.

**Comment by jacobtomlinson** [comment]
Yeah this was merged for the AWS `EC2Cluster` in #369. If someone wants to contribute the same for GCP it should be a good reference.
